readme.md

```diff
@@ -3,71 +3,704 @@
+ return gulp.src('./css/**/*.scss')
+   .pipe(sass().on('error', sass.logError))
+   .pipe(gulp.dest('./css/'))
+});
+```
+
+<details>
+<summary>
+Q. What do we see here in this task thats different?
+</summary>
+<br>
+
+```
+1. sass.logError, which will display an error
+
+2. gulp.dest()
+
+* `gulp.dest` will copies our results to a given directory. In this case to our `css` directory.
+
+```
+<br>
+<br>
+</details>
+
+```bash
+$ gulp sass
+
+```
+
+Great, it worked? But how is that helpful when compiling sass?
+
+### Gulp Watch & Connect
+
+We are going to add these two dependencies as well! Our goal is for us to see any reflected sass changes without manually server reloading or browser refreshing.
+
+Finally, we will be integrating all the tasks to help automate our workflow.
+
+[Gulp Watch](https://www.npmjs.com/package/gulp-watch)
+
+[Gulp Connect](https://www.npmjs.com/package/gulp-connect)
 
-  grunt.loadNpmTasks('grunt-contrib-jshint');
 
+```bash
+$  npm install --save-dev gulp-watch
+
+$ npm install --save-dev gulp-connect
+
+```
+
+```js
+
+var gulp = require('gulp');
+var sass = require("gulp-sass");
+var watch = require('gulp-watch');
+var connect = require('gulp-connect');
+```
+
+```js
+gulp.task('watch', function () {
+ gulp.watch('./css/**/*.scss', ['sass']);
+});
+
+```
+
+```js
+gulp.task('connect', function() {
+  connect.server({
+    livereload: true
+  })
+});
+
+```
+<details>
+<summary>
+Q. What `gulp.watch()` doing ?
+</summary>
+<br>
+
+```
+watching our sass file for changes
+```
+<br>
+<br>
+</details>
+
+
+<details>
+<summary>
+Q. What is `connect.server({livereload: true})` doing?
+</summary>
+<br>
+
+```
+Reloading without manually writing a command in terminal
+```
+<br>
+<br>
+</details>
+
+------
+
+Let's now add `.pipe(connect.reload())` to our `sass` task:
+
+```
+gulp.task('sass', function () {
+ return gulp.src('./css/**/*.scss')
+   .pipe(sass().on('error', sass.logError))
+   .pipe(gulp.dest('./css/'))
+   .pipe(connect.reload());
+});
+
+```
+Finally, let's add the following at the bottom of our `gulpfile.js`:
+
+```
+gulp.task('default', ['sass', 'connect', 'watch']);
+
+```
+
+```bash
+$ gulp
+
+```
+
+<details>
+<summary>
+Q. Why do we not need to specify our task in the above command?
+</summary>
+<br>
+
+```
+Because we used `default`, gulp will automatically execute the `default` command
+```
+<br>
+<br>
+</details>
+
+
+```
+[13:25:59] Using gulpfile ~/wdi8/sandbox/bamsay/gulpfile.js
+[13:25:59] Starting 'sass'...
+[13:25:59] Starting 'connect'...
+[13:25:59] Finished 'connect' after 18 ms
+[13:25:59] Starting 'watch'...
+[13:25:59] Finished 'watch' after 15 ms
+[13:25:59] Server started http://localhost:8080
+[13:25:59] LiveReload started on port 35729
+[13:25:59] Finished 'sass' after 72 ms
+[13:25:59] Starting 'default'...
+[13:25:59] Finished 'default' after 3.63 μs
+
+```
+
+
+## Webpack
+
+[Webpack Documentation](https://webpack.github.io/)
+
+Webpack is a code bundler.
```

Comment:
found this SO to be really useful http://stackoverflow.com/questions/32704827/why-should-one-use-a-module-bundler-webpack-over-a-task-runner-grunt